pkg/agent/manager/cache/cache.go

```diff
@@ -712,16 +712,57 @@ func newSelectorIndex() *selectorIndex {
 	}
 }
 
-func sortIdentities(identities []Identity) {
-	sort.Slice(identities, func(a, b int) bool {
-		return identities[a].Entry.EntryId < identities[b].Entry.EntryId
-	})
-}
-
 func makeIdentity(record *cacheRecord) Identity {
 	return Identity{
 		Entry:      record.entry,
 		SVID:       record.svid.Chain,
 		PrivateKey: record.svid.PrivateKey,
 	}
 }
+
+// sortIdentities returns a slice of identities sorted by SpiffeID.
+//
+// As a stopgap for administratively selected default entries, identities
+// are sorted by path complexity, and alphabetically within path segments.
+//
+// Default order is fewest path segments first in the spirit of more generic
+// identities as default and more complex identies left to workload decision.
+func sortIdentities(identities []Identity) {
+	sort.Sort(byIdentity(identities))
+}
+
+type byIdentity []Identity
+
+func (a byIdentity) Len() int {
+	return len(a)
+}
+
+func (a byIdentity) Swap(i, j int) {
+	a[i], a[j] = a[j], a[i]
+}
+
+func (a byIdentity) Less(i, j int) bool {
+	iSpiffeID, _ := spiffeid.FromString(a[i].Entry.SpiffeId)
```

Comment:
Error handling is intentionally omitted here and after line 748 as the identities have been previously vetted.